src/compiler/crystal/semantic/cleanup_transformer.cr

```diff
@@ -140,6 +140,13 @@ module Crystal
     def transform(node : ClassDef)
       super
 
+      # check superclass for deprecation if the node isn't deprecated
+      if (type = node.type.lookup_type?(node.name)) && !type.annotation(@program.deprecated_annotation)
```

Comment:
BTW: `node.type` is `NilType` ... I must lookup the type by its name (actually a `Path`) to get the actual type for the `ClassDef` node :upside_down_face: 

There's probably a reason, but it escapes me.